control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -3189,8 +3190,21 @@ func (r *HostedControlPlaneReconciler) reconcileOpenShiftAPIServer(ctx context.C
 		}
 	}
 
+	noProxy, err := proxy.DefaultNoProxy(hcp)
+	if err != nil {
+		return fmt.Errorf("failed to calculate noproxy string: %w", err)
+	}
+
+	var imageRegistryAdditionalTrustedCAs *corev1.ConfigMap
+	if hcp.Spec.Configuration != nil && hcp.Spec.Configuration.Image != nil && hcp.Spec.Configuration.Image.AdditionalTrustedCA.Name != "" {
+		imageRegistryAdditionalTrustedCAs = &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: hcp.Spec.Configuration.Image.AdditionalTrustedCA.Name, Namespace: hcp.Namespace}}
+		if err := r.Get(ctx, client.ObjectKeyFromObject(imageRegistryAdditionalTrustedCAs), imageRegistryAdditionalTrustedCAs); err != nil {
```

Comment:
is this deployment reconciliation driven by this configMap or would this introduce a race?

Reply:
It is driven by the configmap which comes from the higher level HostedCluster API